test/common_utils.py

```diff
@@ -270,84 +270,57 @@ def combinations_grid(**kwargs):
     return [dict(zip(kwargs.keys(), values)) for values in itertools.product(*kwargs.values())]
 
 
-class ImagePair(TensorLikePair):
-    def __init__(
-        self,
-        actual,
-        expected,
-        *,
-        mae=False,
-        **other_parameters,
-    ):
-        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
-            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
+assert_equal = functools.partial(torch.testing.assert_close, rtol=0, atol=0)
 
-        super().__init__(actual, expected, **other_parameters)
-        self.mae = mae
 
-    def compare(self) -> None:
-        actual, expected = self.actual, self.expected
+def assert_close_with_image_support(actual, expected, *, mae=False, atol=None, msg=None, **kwargs):
+    def compare(actual, expected):
+        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
+            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
 
-        self._compare_attributes(actual, expected)
-        actual, expected = self._equalize_attributes(actual, expected)
+        enable_mae_comparison = all(isinstance(input, torch.Tensor) for input in [actual, expected])
```

Comment:
Should this be

```suggestion
        enable_mae_comparison = all(isinstance(input, torch.Tensor) for input in [actual, expected]) and mae
```

? Regardless I don't really understand the reason for `enable_mae_comparison`. Why can't we just convert all inputs to tensors and apply `mae` comparison iff `mae = True`?

Reply:
I wrote this as a general function that one can use like `assert_close`, but `with_image_support`. Meaning, you can still do `assert_close_with_image_support(None, 5)` and get a proper error message. If we force everything into a tensor, that would not be possible.

Reply:
But are we ever going to use it when the inputs aren't images (either PIL or Tensors)?

Reply:
Yes, we have such usages:

- In the functional v2 tests, we are using `assert_close` (or now `assert_close_with_image_support`) for all kernels and dispatchers and this of course also means bounding boxes, masks and videos.
- In the consistency tests for the detection and segmentation tests, we test full samples against each other and thus the inputs are more than just images. This is somewhat also the case in our (fairly limited) transforms tests, but will probably increase there when we ramp them up.